static/app/components/forms/fields/accessibility.spec.tsx

```diff
@@ -0,0 +1,196 @@
+import TextareaField from './textareaField';
+import TextField from './textField';
+
+describe('Field accessibility', function () {
+  it('has appropriate aria attributes on all fields', async function () {
+    // TODO(epurkhiser): The following fields are sill missing accessibility
+    // check tests:
+    //
+    // - ChoiceMapper
+    // - ProjectMapperField
+    // - SentryProjectSelectorField
+    // - TableField
+    // - DatePickerField
+    // - DateTimeField
+    // - FileField
+
+    // TODO(epurkhiser): It would be really nice if we could enforce that every
+    // field that exists in `components/forms/fields/*` has a proper
+    // accessibility test here.
+
+    const model = new FormModel();
+
+    render(
+      <Form model={model}>
+        <SeparatorField />
+        <HiddenField name="hidden" defaultValue="itsHidden" />
+        <TextField label="My Text Input" help="This is a text input" name="myTextInput" />
+        <NumberField
+          label="My Number Input"
+          help="This is a number input"
+          name="myNumberInput"
+        />
+        <EmailField
+          label="My Email Input"
+          help="This is a email input"
+          name="myEmailInput"
+        />
+        <SecretField
+          label="My Password"
+          help="This is a password input"
+          name="mySecretInput"
+        />
+        <TextareaField
+          label="My Textarea"
+          help="This is a text area input"
+          name="myTextarea"
+        />
+        <CheckboxField
+          label="My Checkbox"
+          help="This is a checkbox (not a switch)"
+          name="myCheckbox"
+        />
+        <BooleanField
+          label="My Boolean"
+          help="This is a boolean switch toggle"
+          name="myBoolean"
+        />
+        <RadioField
+          label="My Radios"
+          choices={[
+            ['thing_1', 'Thing 1', 'Thing 1 description'],
+            ['thing_2', 'Thing 2', 'Thing 2 description'],
+          ]}
+          help="This is a radio set field"
+          name="myRadios"
+        />
+        <RangeField
+          label="My Range Slider"
+          min={0}
+          max={100}
+          step={10}
+          defaultValue={10}
+          help="This is a range slider"
+          name="myRangeSlider"
+        />
+        <SelectField
+          label="My Select"
+          options={[
+            {value: 'item1', label: 'Item 1'},
+            {value: 'item2', label: 'Item 2'},
+          ]}
+          help="This is a select field field"
+          name="mySelectbox"
+        />
+      </Form>
+    );
+
+    // Separator field
+    expect(screen.getByRole('separator')).toBeInTheDocument();
+
+    // Hidden field
+    expect(model.getValue('hidden')).toBe('itsHidden');
+
+    // Text Input
+    const textInput = screen.getByRole('textbox', {name: 'My Text Input'});
+    userEvent.type(textInput, 'testing');
+    expect(textInput).toHaveValue('testing');
```

Comment:
shouldn't we `query` the field again after changing its value? I found myself wanting to do the same a couple of times but was afraid this would cause some issues

Reply:
This seems fine. I haven't seen any issues.